src/Lumi/Components2/Text.purs

```diff
@@ -0,0 +1,270 @@
+module Lumi.Components2.Text
+  ( Text
+  , TextProps, TextType
+  , text
+
+  , TextModifier
+  , body, strong, emphasized, subtext
+  , muted, color
+
+  , ParagraphProps
+  , paragraph_, paragraph
+
+  , Header
+  , HeaderProps
+  , subsectionHeader
+  , sectionHeader
+  , title
+  , mainHeader
+  ) where
+
+import Prelude
+
+import Data.Maybe (Maybe(..), fromMaybe, maybe)
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, PropsModifier, PropsModifier', lumiComponent, propsModifier)
+import Lumi.Components.Color (ColorMap)
+import Lumi.Styles (Style, StyleModifier, StyleProperty)
+import Lumi.Styles as S
+import Lumi.Styles.Theme (LumiTheme(..), TextMap, textFontSize, textLineHeight, textMargin, useTheme)
+import React.Basic (JSX, ReactComponent)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Hooks as Hooks
+
+-- Text
+
+data Text = Text
+
+data TextType
+  = Body
+  | Strong
+  | Emphasis
+  | Subtext
+
+type TextProps =
+  ( component :: Text
+  , type :: Maybe TextType
+  , content :: String
+  )
+
+type TextElement = ReactComponent { children :: Array JSX, className :: String }
+
+-- | Inline piece of text whose `content` prop is a `String`. Elements of this
+-- | component should preferentially be used only where an inline element is
+-- | expected, that is, from within the contents of a block-level element.
```

Comment:
I'm a bit confused about this; can't anything go within the contents of a block-level element?

Reply:
Yeah, this is poorly worded, and potentially false I guess. I'll try rephrasing or simplifying it.

Reply:
What do you think of it now?
```purescript
-- | Inline piece of text whose `content` prop is a `String`. Elements of this
-- | component should preferentially be used only where an inline piece of text
-- | with no bottom margin is expected.
-- |
-- | For block-level, independent text elements, please use `paragraph`,
-- | `paragraph_`, or the header components defined in this module.
text :: LumiComponent TextProps
```

Reply:
Looks good :+1: